### PR TITLE
impbook IPR report v0

### PR DESCRIPTION
**Pull request by iaincrawf**
Can't figure out how to get the impbook Id into the href to link to the import book ut, because I need the rowId to be a combo of it and orderDetails.lineNumber so that it's unique and shows all rows.

Also currently can't get impbook ut on apps so figure I missed something routing wise but can't see what, have made a change to the impbook module which may or may not help

### CODE REVIEW

File: src/Domain.LinnApps/Reports/ImportBookReportService.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        }
+
+        public ResultsModel GetIPRReport(DateTime from, DateTime to)
+        {
+            var iprImpbooks = this.impbookRepository.FilterBy(
+                x =>
+
+                    // x.IprCpcNumber.HasValue && x.IprCpcNumber.Value == 13 &&
+                    // todo check if I can just use that ^, not sure it'll be populated reliably 
+                    x.OrderDetails.Any(z => z.CpcNumber.HasValue && z.CpcNumber.Value == this.IprCpcNumberId)
```
Comment (andrewfraser73) [inline src/Domain.LinnApps/Reports/ImportBookReportService.cs:33]: as discussed before 13 might work but not necessarily be future proof.  it gets us going. The most probably mistake that will happen is Rhona or someone will pick the CPC number that is close but wrong (I can see 2 this year using the extra 0 version) however all my processing will also only really work with 13

File: src/Facade/Services/ImportBookReportReportFacadeService.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            DateTime from;
+            DateTime to;
+            try
+            {
+                from = DateTime.Parse(resource.FromDate);
+                to = DateTime.Parse(resource.ToDate);
+            }
+            catch (Exception)
+            {
+                return new BadRequestResult<ResultsModel>("Invalid dates supplied to assembly fails details report");
```
Comment (andrewfraser73) [inline src/Facade/Services/ImportBookReportReportFacadeService.cs:34]: think this must be a copy paste from assembly fails as should prob say impbook ipr report

File: src/Service.Host/client/src/components/reports/ImpbookIprReport.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+const handleBackClick = (history, options) => {
+    const uri = `/logistics/import-books/ipr?fromDate=${encodeURIComponent(
+        options.fromDate
+    )}&toDate=${encodeURIComponent(options.toDate)}`;
+
+    history.push(uri);
+};
+
+const AssemblyFailsDetails = ({ reportData, loading, history, options, config }) => (
```
Comment (andrewfraser73) [inline src/Service.Host/client/src/components/reports/ImpbookIprReport.js:20]: lots of references to assemblyfails details might want to find/replace them to impbookipr

File: src/Service.Host/client/src/components/reports/ImpbookIprReport.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                )}
+            </Grid>
+            <Grid item xs={12}>
+                <BackButton backClick={() => handleBackClick(history, options)} />
+            </Grid>
+        </Grid>
+    </Page>
+);
+
+AssemblyFailsDetails.propTypes = {
```
Comment (andrewfraser73) [inline src/Service.Host/client/src/components/reports/ImpbookIprReport.js:55]: same as before

**Review by andrewfraser73 — APPROVED** [review]
might want to change a couple of the assemblyfail refs

rest looks alright but might double check the report results when its live on production

File: src/Domain.LinnApps/Reports/ImportBookReportService.cs (hunk 1 above)
Comment (iaincrawf) [reply src/Domain.LinnApps/Reports/ImportBookReportService.cs:33]: Yeah, we could pull all of the cpc numbers in easily enough but even then, we either use the right one with id 13, or include the other similar ones and maybe get dodgy results! Maybe I can put an 'are you sure this is the right cpc number/ is this IPR?' popup in the new form to hopefully stop the wrong one being used

File: src/Facade/Services/ImportBookReportReportFacadeService.cs (hunk 2 above)
Comment (iaincrawf) [reply src/Facade/Services/ImportBookReportReportFacadeService.cs:34]: Yeah sorry, lots of copy and pasting for speed - should have checked through it a bit better though!

File: src/Domain.LinnApps/Reports/ImportBookReportService.cs (hunk 1 above)
Comment (andrewfraser73) [reply src/Domain.LinnApps/Reports/ImportBookReportService.cs:33]: i would just leave it is until we get the report working.  it is a future worry that doesn't stop you doing what you need to do

**Pull request merged**